pkg/config/config.go

```diff
@@ -30,17 +37,39 @@ func Proxy() (*pb.Proxy, error) {
 func unmarshalConfig(filepath string, msg proto.Message) error {
 	configJSON, err := ioutil.ReadFile(filepath)
 	if err != nil {
-		log.Errorf("error reading %s: %s", filepath, err)
-		return err
+		return fmt.Errorf("Error reading config: %s", err)
 	}
 
 	log.Debugf("%s config JSON: %s", filepath, configJSON)
 
 	err = unmarshaler.Unmarshal(bytes.NewReader(configJSON), msg)
 	if err != nil {
-		log.Errorf("error unmarshaling %s: %s", filepath, err)
-		return err
+		return fmt.Errorf("Error unmarshaling config: %s", err)
 	}
 
 	return nil
 }
+
+// Fetch the configuration from the kubernetes API.
+func Fetch(k corev1.ConfigMapInterface) (global *pb.Global, proxy *pb.Proxy, err error) {
```

Comment:
Curious as to which consumers need this? We are hoping that all the consumer pods can just mount the config map and rely on the `Global()` and `Proxy()` functions above to get access to the configs.

Reply:
The CLI needs access to the cluster's configuration. Doing so during inject unblocks [identity work](https://github.com/linkerd/linkerd2/pull/new/ver%2Fidentity-inject); but even after inject is moved into a controller, install will need to be able to inspect the cluster's configuration. Any tooling to manage identity configuration will also need to be able to fetch the global config.

Reply:
:ok_hand: Just wanna confirm whether this is a stop-gap solution, or something that we will want to keep around. Sounds like the latter.